### PR TITLE
Jb oauth fixes

### PR DESCRIPTION
**Pull request by jbasdf**

### CODE REVIEW

File: db/migrate/20210501034609_add_legacy_lti_user_id_to_users.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddLegacyLtiUserIdToUsers < ActiveRecord::Migration[6.1]
+  def change
+    add_column :users, :legacy_lti_user_id, :string
```
Comment (bfcoder) [inline db/migrate/20210501034609_add_legacy_lti_user_id_to_users.rb:3]: As you are doing a database lookup based on this, do you want an index on it?

File: db/seeds.rb [hunk 2]
```diff
@@ -187,7 +187,6 @@
         oidc_url: "https://blackboard.com/",
       },
     ],
-
```
Comment (bfcoder) [inline db/seeds.rb:190]: Revert this change so it doesn't cause downstream merging headaches?

File: app/controllers/concerns/lti_support.rb [hunk 3]
```diff
@@ -144,6 +145,7 @@ def _generate_new_lti_user(params)
     user.password = ::SecureRandom::hex(15)
     user.password_confirmation = user.password
     user.lti_user_id = lti_user_id
+    user.legacy_lti_user_id = lti_user_id
```
[the remark this answers is not in the document]
Comment (jbasdf) [reply app/controllers/concerns/lti_support.rb:148]: Canvas only sends the 1.1 lti_user_id during an OAuth request. If they change that we'd have to change it as well but if they do it would break a lot of tools so I don't know that they will change it anytime soon. I'm guessing we'll need this for years to come.

File: db/seeds.rb (hunk 2 above)
Comment (jbasdf) [reply db/seeds.rb:190]: done

File: db/migrate/20210501034609_add_legacy_lti_user_id_to_users.rb (hunk 1 above)
Comment (jbasdf) [reply db/migrate/20210501034609_add_legacy_lti_user_id_to_users.rb:3]: Yes

**Review by bfcoder — APPROVED** [review]
Looks good
